### PR TITLE
[fix]: [CCM-8233]: Recover network failure (Create watcher again)

### PR DESCRIPTION
**Pull request by sdsingh07**
## Describe your changes
Added listPodForAllNamespacesCall logs for testing. And, handle network failure.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35294)

### CODE REVIEW

**Review by hitesharinga — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35294)* status: :shipit: complete! ✓

**Comment by sdsingh07** [comment]
trigger messagecheck

File: 930-delegate-tasks/src/main/java/io/harness/perpetualtask/k8s/utils/K8sWatcherHelper.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class K8sWatcherHelper {
+  public static final String POD_WATCHER_PREFIX = "PodWatcher-%s";
+  public static final String NODE_WATCHER_PREFIX = "NodeWatcher-%s";
+  public static final String PV_WATCHER_PREFIX = "PVWatcher-%s";
+  private static final long WAIT_TIME = Duration.ofMinutes(10).getSeconds();
```
Comment (hitesharinga) [inline 930-delegate-tasks/src/main/java/io/harness/perpetualtask/k8s/utils/K8sWatcherHelper.java:25]: 45 mins 
create a count and add condition < 3 for count 
-- > delete  
--> how to we stop the informer. (podwatcher and nodewatcher )
--> remove pv from check. add check for pod, node and count

**Review by hitesharinga — APPROVED** [review]
> Reviewed 6 of 6 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35294)* status: :shipit: complete! ✓

**Pull request merged**